### PR TITLE
React at login

### PR DESCRIPTION
**Pull request by alesshh**
### Description
Setup react configurations at webpack and refactor login page to use it.

### How to test?
Login page must be working

PS: I will setup react tests with enzyme in a separated.

### CODE REVIEW

File: frontend/pages/login/login-button/login-button.css [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.gButton {
```
Comment (dvinciguerra) [inline frontend/pages/login/login-button/login-button.css]: Aqui não poderia ser utilizado css modules para evitar conflito e explorar o desacoplamento dos componentes para o reaproveitamento (talvez)!?

```css
/* frontend/pages/login/login-button/login-button.module.css */
.gButton {
    ...
}
```
E na hora de utilizar...

```javascript
import styles from './login-button.module.css';

// no render do controller
render() {
  return (
      <div className="row justify-content-center align-items-center p-3">
        <div ref={this.buttonContainerRef} className={styles.gButton} />
      </div>
    );
  }
```
Ref.: https://facebook.github.io/create-react-app/docs/adding-a-css-modules-stylesheet

Comment (alesshh) [reply frontend/pages/login/login-button/login-button.css]: Boa! A princípio só fiz split do css, mas acho que já vale a pena sim!

Comment (dvinciguerra) [reply frontend/pages/login/login-button/login-button.css]: Boaaa, do mais achei massa demais!
Congrats! =D

**Comment by alesshh** [comment]
@juliemar @lccezinha @dvinciguerra 

(PNG Image, 225 × 225 pixels)

File: frontend/pages/login/logo/index.js [hunk 2]
```diff
@@ -0,0 +1 @@
+export { Logo } from "./logo";
```
Comment (lccezinha) [inline frontend/pages/login/logo/index.js]: Why we need a `logo/index.js` and a `logo/logo.js`? What the diff between import the logo/index and logo/logo file?

Comment (alesshh) [reply frontend/pages/login/logo/index.js]: `logo/index.js` is for simplify imports like `import logo from './logo'` instead `import logo from './logo/logo`.  We can move the logo source for `index` file and reduce the number of files, but, an problem with this approach is that, when editing multiple Components at once in your editor, the tabs are all labeled index.js.
What you think about?

**Comment by alesshh** [comment]
> Made a comment about the code.
> 
> I do not have enough knowledge to review the webpack configs.
> 
> Also when I tried to run the app locally, I got this error:
> 
> image

What command that you using to run  project?

File: frontend/pages/login/logo/index.js (hunk 2 above)
Comment (alesshh) [reply frontend/pages/login/logo/index.js]: Maybe we can reduce the number of files for now, until this becomes a problem.

Comment (lccezinha) [reply frontend/pages/login/logo/index.js]: No problem about that just was trying to understand the `why` behind this approach. Your explanation makes all sense, for me is good to go this way (with index files for import and component.file with real code).

About the error when trying to run the project; I use the same command that I was used to `docker-compose up`. :eyes:

**Comment by alesshh** [comment]
@juliemar can you confirm that the problem  @lccezinha had happens with you?
In my machine, a clean `docker-compose up` works  fine.


My log
```
› docker-compose up
Building web
Step 1/11 : FROM node:10.15.3-alpine
 ---> d7c77d094be1
Step 2/11 : RUN mkdir -p /var/app
 ---> Using cache
 ---> 13e288827d0f
Step 3/11 : WORKDIR /var/app
 ---> Using cache
 ---> a503c8460c70
Step 4/11 : COPY docker-entrypoint.sh /docker-entrypoint.sh
 ---> 0bd4d512d95a
Step 5/11 : COPY . /var/app
 ---> 59f416c33419
Step 6/11 : EXPOSE 8080
 ---> Running in f46c48727ae4
Removing intermediate container f46c48727ae4
 ---> c0d2e3821bef
Step 7/11 : RUN npm install
 ---> Running in 615fcd706236

> nodemon@1.18.10 postinstall /var/app/node_modules/nodemon
> node bin/postinstall || exit 0

Love nodemon? You can now support the project via the open collective:
 > https://opencollective.com/nodemon/donate


> webpack-cli@3.3.1 postinstall /var/app/node_modules/webpack-cli
> node ./bin/opencollective.js



                            Thanks for using Webpack!
                 Please consider donating to our Open Collective
                        to help us maintain this package.



                 Donate: https://opencollective.com/webpack/donate


npm WARN matrix@1.0.0 No repository field.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.8 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.8: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

added 1054 packages from 935 contributors and audited 13492 packages in 24.426s
found 0 vulnerabilities

Removing intermediate container 615fcd706236
 ---> f0902cbe0235
Step 8/11 : RUN npm run build-backend
 ---> Running in 720226edbc36

> matrix@1.0.0 build-backend /var/app
> babel ./backend -d dist

Successfully compiled 6 files with Babel.
Removing intermediate container 720226edbc36
 ---> f9db775347e7
Step 9/11 : RUN npm run build-frontend
 ---> Running in 4ffd19392faf

> matrix@1.0.0 build-frontend /var/app
> webpack --mode production --config webpack.production.js

(node:16) DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead

WebpackCleanupPlugin: 4 file(s) deleted.
Hash: cebc182349b62553e438
Version: webpack 4.30.0
Time: 11803ms
Built at: 06/17/2019 7:38:20 PM
                          Asset       Size  Chunks                    Chunk Names
  login-07f493ee598b05a6733e.js    124 KiB       0  [emitted]         login
 login-e1d65a2e6b1a3c5fd862.css    151 KiB       0  [emitted]         login
                  manifest.json  198 bytes          [emitted]
office-1a985c7b42c353c5a1b4.css    154 KiB       1  [emitted]         office
 office-cbf20629c48c41fc1b55.js    281 KiB       1  [emitted]  [big]  office
Entrypoint office [big] = office-1a985c7b42c353c5a1b4.css office-cbf20629c48c41fc1b55.js
Entrypoint login [big] = login-e1d65a2e6b1a3c5fd862.css login-07f493ee598b05a6733e.js
 [7] (webpack)/buildin/global.js 472 bytes {0} {1} [built]
 [8] ./frontend/profile.js 1.16 KiB {0} {1} [built]
[26] ./frontend/pages/login/footer/footer.module.css 127 bytes {0} [built]
[41] ./frontend/office-events.js 1.96 KiB {1} [built]
[43] ./frontend/header.js 659 bytes {1} [built]
[46] ./frontend/pages/login/login-button/login-button.module.css 95 bytes {0} [built]
[47] ./frontend/pages/login/login.module.css 97 bytes {0} [built]
[48] ./frontend/office.js 8.14 KiB {1} [built]
[49] ./frontend/office.css 39 bytes {1} [built]
[75] ./frontend/header.css 39 bytes {1} [built]
[79] ./frontend/context-menu/context-menu.js 1.38 KiB {1} [built]
[80] ./frontend/initialize/jquery.initialize.js 1.85 KiB {1} [built]
[88] ./frontend/pages/login/global.css 39 bytes {0} [built]
[89] ./frontend/login-application.js + 10 modules 8.91 KiB {0} [built]
     | ./frontend/login-application.js 183 bytes [built]
     | ./frontend/pages/login/index.js 32 bytes [built]
     | ./frontend/pages/login/login.js 3.85 KiB [built]
     | ./frontend/pages/login/footer/index.js 34 bytes [built]
     | ./frontend/pages/login/login-button/index.js 48 bytes [built]
     | ./frontend/pages/login/logo/index.js 30 bytes [built]
     | ./frontend/pages/login/title/index.js 32 bytes [built]
     | ./frontend/pages/login/footer/footer.js 402 bytes [built]
     | ./frontend/pages/login/login-button/login-button.js 3.78 KiB [built]
     | ./frontend/pages/login/logo/logo.js 235 bytes [built]
     | ./frontend/pages/login/title/title.js 275 bytes [built]
[92] ./node_modules/@sentry/browser/esm/sdk.js + 32 modules 143 KiB {1} [built]
     |    33 modules
    + 92 hidden modules

WARNING in asset size limit: The following asset(s) exceed the recommended size limit (244 KiB).
This can impact web performance.
Assets:
  office-cbf20629c48c41fc1b55.js (281 KiB)

WARNING in entrypoint size limit: The following entrypoint(s) combined asset size exceeds the recommended limit (244 KiB). This can impact web performance.
Entrypoints:
  office (435 KiB)
      office-1a985c7b42c353c5a1b4.css
      office-cbf20629c48c41fc1b55.js
  login (275 KiB)
      login-e1d65a2e6b1a3c5fd862.css
      login-07f493ee598b05a6733e.js


WARNING in webpack performance recommendations:
You can limit the size of your bundles by using import() or require.ensure to lazy load some parts of your application.
For more info visit https://webpack.js.org/guides/code-splitting/
Child mini-css-extract-plugin node_modules/css-loader/dist/cjs.js??ref--4-1!frontend/header.css:
    Entrypoint mini-css-extract-plugin = *
    [0] ./node_modules/css-loader/dist/cjs.js??ref--4-1!./frontend/header.css 705 bytes {0} [built]
        + 1 hidden module
Child mini-css-extract-plugin node_modules/css-loader/dist/cjs.js??ref--4-1!frontend/office.css:
    Entrypoint mini-css-extract-plugin = *
    [0] ./node_modules/css-loader/dist/cjs.js??ref--4-1!./frontend/office.css 3.68 KiB {0} [built]
        + 1 hidden module
Child mini-css-extract-plugin node_modules/css-loader/dist/cjs.js??ref--4-1!frontend/pages/login/global.css:
    Entrypoint mini-css-extract-plugin = *
    [0] ./node_modules/css-loader/dist/cjs.js??ref--4-1!./frontend/pages/login/global.css 383 bytes {0} [built]
        + 1 hidden module
Child mini-css-extract-plugin node_modules/css-loader/dist/cjs.js??ref--4-1!node_modules/bootstrap/dist/css/bootstrap.css:
    Entrypoint mini-css-extract-plugin = *
       2 modules
Child mini-css-extract-plugin node_modules/css-loader/dist/cjs.js??ref--5-1!frontend/pages/login/footer/footer.module.css:
    Entrypoint mini-css-extract-plugin = *
    [0] ./node_modules/css-loader/dist/cjs.js??ref--5-1!./frontend/pages/login/footer/footer.module.css 534 bytes {0} [built]
        + 1 hidden module
Child mini-css-extract-plugin node_modules/css-loader/dist/cjs.js??ref--5-1!frontend/pages/login/login-button/login-button.module.css:
    Entrypoint mini-css-extract-plugin = *
    [0] ./node_modules/css-loader/dist/cjs.js??ref--5-1!./frontend/pages/login/login-button/login-button.module.css 282 bytes {0} [built]
        + 1 hidden module
Child mini-css-extract-plugin node_modules/css-loader/dist/cjs.js??ref--5-1!frontend/pages/login/login.module.css:
    Entrypoint mini-css-extract-plugin = *
    [0] ./node_modules/css-loader/dist/cjs.js??ref--5-1!./frontend/pages/login/login.module.css 330 bytes {0} [built]
        + 1 hidden module


                            Thanks for using Webpack!
                 Please consider donating to our Open Collective
                        to help us maintain this package.



                 Donate: https://opencollective.com/webpack/donate


Removing intermediate container 4ffd19392faf
 ---> 29dfdef99f80
Step 10/11 : ENTRYPOINT ["sh","/docker-entrypoint.sh"]
 ---> Running in ade82df68dea
Removing intermediate container ade82df68dea
 ---> 3e0e53574f04
Step 11/11 : CMD ["npm" , "run", "start-backend"]
 ---> Running in 5bd9ab034755
Removing intermediate container 5bd9ab034755
 ---> 080679a2c34e
Successfully built 080679a2c34e
Successfully tagged matrix_web:latest
WARNING: Image for service web was built because it did not already exist. To rebuild this image you must use `docker-compose build` or `docker-compose up --build`.
Creating matrix_web_1 ... done
Attaching to matrix_web_1
web_1  |
web_1  | > matrix@1.0.0 start /var/app
web_1  | > npm-run-all -p start-backend-development-server start-frontend-development-server
web_1  |
web_1  |
web_1  | > matrix@1.0.0 start-backend-development-server /var/app
web_1  | > nodemon --exec babel-node ./backend/app.server.js
web_1  |
web_1  |
web_1  | > matrix@1.0.0 start-frontend-development-server /var/app
web_1  | > webpack --watch --config webpack.development.js
web_1  |
web_1  | [nodemon] 1.18.10
web_1  | [nodemon] to restart at any time, enter `rs`
web_1  | [nodemon] watching: *.*
web_1  | [nodemon] starting `babel-node ./backend/app.server.js`
web_1  |
web_1  | webpack is watching the files…
web_1  |
web_1  | (node:56) DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead
web_1  | Running on http://0.0.0.0:8080
web_1  | [ { id: '00131978-04c4-45ee-991e-2e2395af1d97',
web_1  |     name: 'The Dock',
web_1  |     disableMeeting: true },
web_1  |   { id: '8cd461e7-c257-492a-9136-c49764d08d3d',
web_1  |     name: 'Nebuchadnezzar' },
web_1  |   { id: 'd627d576-494e-47de-b8c5-96453c8e686c', name: 'Logos' },
web_1  |   { id: '3f594b32-f37b-44e3-ab58-76cc969b7ed9', name: 'Osiris' },
web_1  |   { id: '1c3ffab3-3931-4de7-86fe-5e1f882acf90', name: 'Icarus' },
web_1  |   { id: 'ee7667b1-3a6a-4f1d-ae38-5d9932bfb118', name: 'Caduceus' },
web_1  |   { id: '1e5d4678-ad76-4be6-ad39-fdada949b5ab', name: 'Brahma' },
web_1  |   { id: 'a6c60a33-6943-473d-836a-6d33e7f08d9d', name: 'Novalis' },
web_1  |   { id: 'de5fac4f-8622-46f7-8f0f-0c02a68ca846', name: 'Vigilant' },
web_1  |   { id: 'e8b2c565-e718-4d40-bf10-51b0537b8f37', name: 'Zion' } ]
web_1  | socket.id: IrY5qN-42Be4Ph_8AAAA
web_1  | connected: 110446043137341152318
```

**Pull request merged**